libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -0,0 +1,396 @@
+use super::*;
+
+use glam::DVec2;
+
+impl Subpath {
+	/// Calculate the point on the subpath based on the parametric `t`-value provided.
+	/// Expects `t` to be within the inclusive range `[0, 1]`.
+	pub fn evaluate(&self, t: ComputeType) -> DVec2 {
+		match t {
+			ComputeType::Parametric { t } => {
+				assert!((0.0..=1.).contains(&t));
+
+				let number_of_curves = self.len_segments() as f64;
+				let scaled_t = t * number_of_curves;
+
+				let target_curve_index = scaled_t.floor() as i32;
+				let target_curve_t = scaled_t % 1.;
+
+				if let Some(curve) = self.iter().nth(target_curve_index as usize) {
+					curve.evaluate(target_curve_t)
+				} else {
+					self.iter().last().unwrap().evaluate(1.)
```

Comment:
The `nth` function will return the nth element of the iterator, with indexing starting at 0. For any `t < 1`, the `target_curve_index` would round down to something `< number_of_curves` and hence actually return a curve when `nth` is called. This clause handles the case where `t = 1.` (aka return `t = 1` for the very last curve in the subpath).

Reply:
Yup, what happens is that when `t=1` for a bezier segment at index `i`, it will actually return the point `t=0` for the bezier segment `i+1`. This works well until we reach the last curve, in which case we would index out of bounds, that's what the else clause is checking. When `t=1` for the subpath, we want `t=1` for the last bezier segment.